sidekick_core/lib/sidekick_core.dart

```diff
@@ -320,24 +322,16 @@ class SidekickCommandRunner<T> extends CompletionCommandRunner<T> {
       }
 
       try {
-        if (_isUpdateCheckEnabled &&
-            !_isSidekickCliUpdateCommand(parsedArgs) &&
-            !isRunningTabCompletionCommand()) {
-          // print warning if the user didn't fully update their CLI
-          _checkCliVersionIntegrity();
-          // print warning if CLI update is available
-          // TODO start the update check in the background at command start
-          // TODO prevent multiple update checks when a command start another command
-          await _checkForUpdates();
-        }
+        (await updateCheck)?.call();
       } finally {
         unmount();
       }
     }
   }
 
   /// Print a warning if the CLI isn't up to date
-  Future<void> _checkForUpdates() async {
+  Future<void Function()> _checkForUpdates() async {
+    void Function()? updateCheck;
     try {
       final updateFuture = VersionChecker.isDependencyUpToDate(
```

Comment:
The try catch is now useless, because the `updateFuture` is not awaited

Reply:
The Future itself is awaited in Line 345 in combination with the timeout
Errors are caught and result in returning null